packages/react-next/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+import { classNamesFunction, mergeAriaAttributeValues, warnMutuallyExclusive, FocusRects } from '../../Utilities';
+import { Icon } from '../../Icon';
+import { ICheckboxProps, ICheckboxStyleProps, ICheckboxStyles } from './Checkbox.types';
+import { KeytipData } from '../../KeytipData';
+import { useId, useControllableValue } from '@uifabric/react-hooks';
+
+const getClassNames = classNamesFunction<ICheckboxStyleProps, ICheckboxStyles>();
+
+export const CheckboxBase = React.memo(
```

Comment:
why are you using `React.memo` here? whether to memoize should be determined upon using `checkbox` (mostly in app code) but not at `checkbox` level (component library)

Reply:
I can't say I agree. Why shouldn't we prevent rerenders if the props don't change?

Reply:
https://twitter.com/dan_abramov/status/1083897065263034368
memoization comes at a cost. we should memoize only if it's solving a real perf issue. and that decision should be made when Checkbox is being used, not as default behavior we enforce

Reply:
As [this reply to that tweet] points out, there's no memory increase (React stores the old props anyway), and we're comparing a small set of props.

Reply:
as Dan's reply: the idea of small-but-accumulating cost when you do it everywhere is similar.
I still think we shouldn't make such decision for users. what if user want to use other ways of comparison instead of default (shallow) comparison

Reply:
I fundamentally disagree. 
* Apps don't/shouldn't know how Checkbox is implemented, so they'd have no reason to implement their own comparison operator.
* As the argument goes, JS is fast. Running a shallow comparison against props that were already being held in memory anyway is minor. This comparison isn't run for the first render, so the question is which is more likely?
  * The accumulated cost of these shallow compares during a rerender across a sufficiently large number of components causes noticeable perf degradation, or
  * The shallow comparison prevents enough (more expensive) rerenders to prevent a perf degredation.
  * It makes no noticeable impact on perf at all.

My money is on one of the latter two.

Lastly, as an app author, I don't want to have to wrap imported components in `React.memo()`, especially if I would have to do it across multiple components. Maintaining that would be a massive headache.

Reply:
Let's take a step back and ask ourselves when we should memo a (any) component:
1. Renders often
2. Re-renders with the same props (by shallow or deep or other custom comparison)
3. Component render is expensive

Both 1 & 2 are controlled by App (user's) code

If it's 3, then I think we (component library) should do one of the following:
1. why is render expensive? make it better
2. consider apply memo by default if we have bad perf (in this case, Nav/DetailsList can be such candidate)

For this particular case - Checkbox - we don't see bad perf on re-renders (https://codesandbox.io/s/festive-architecture-nfe33) with same props, so why do we need to apply default memo on it?